workspaces/ui-v2/src/pages/docs/components/DataFetchers.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC } from 'react';
+import { createFlatList } from '<src>/components/FieldOrParameter';
+import { useShapeDescriptor } from '<src>/hooks/useShapeDescriptor';
+import { IShapeRenderer } from '<src>/components';
+
+type ContributionFetcherProps = {
+  rootShapeId: string;
+  endpointId: string;
+  // TODO QPB change this typing - currently this holds contributions, and the data for rendering the shape
+  children: (
+    contributions: {
+      id: string;
+      contributionKey: string;
+      value: string;
+      endpointId: string;
+      depth: number;
+      name: string;
+      shapes: IShapeRenderer[];
+    }[]
+  ) => React.ReactNode;
+};
+
+// TODO QPB replace this by fetching the contributions in redux
```

Comment:
A progression towards having more generic, composable components. What needs to happen next is:
- Build these fetchers for changelog  + diff pages + other components
- Migrate shapes / etc into redux
- rip out these data adaptors and pass in data directly from redux